run/order.go

```diff
@@ -157,9 +158,26 @@ func BuildDAG(
 
 	visited[dag.ID(s.Dir.String())] = struct{}{}
 
-	removeWrongPaths := func(fieldname string, paths []string) []string {
-		cleanpaths := []string{}
+	computePaths := func(fieldname string, paths []string) ([]string, error) {
+		uniqPaths := map[string]struct{}{}
 		for _, pathstr := range paths {
+			if strings.HasPrefix(pathstr, "tag:") {
+				if fieldname != "before" && fieldname != "after" {
+					return nil, errors.E(
+						"tag:<query> is not allowed in %q field", fieldname,
```

Comment:
@mariux this is where we disallow `tag:` filters in `wants` and `wanted_by`.

Reply:
nice... :+1: